### DISCUSSION TITLE
Need support

### DISCUSSION THREAD

**Discussion by psidana1983** [camaraproject/DeviceStatus#48] [Q&A]
Hi Team , 
Can you please help me to understand what is the significance of the request fields
- ipv6Addr
- ipv4 
- external Id 

I understand that these are the mandatory request parameter but we don't understand why it is required. 
I checked with my team and as per the feedback we can fetch the details based on mobile number itself . So want to understand if it is mandatory to implement it  and how does the response change based on these fields.

**Comment by eric-murray** [comment]
@psidana1983
The `UeId` schema is currently defined as:
```
    UeId:
      description: User equipment identifier
      type: object
      properties:
        externalId:
          $ref: '#/components/schemas/ExternalId'
        msisdn:
          $ref: '#/components/schemas/MSISDN'
        ipv4Addr:
          $ref: '#/components/schemas/Ipv4Addr'
        ipv6Addr:
          $ref: '#/components/schemas/Ipv6Addr'
      minProperties: 1
```
So only one of these parameters is required. If you have `msisdn` (i.e. the mobile phone number), then that is all that is required.

Note that the current API definition is pre-release, and this schema will be updated in the not too distant future following changes likely to be approved within other CAMARA sub-projects.

**Comment by eric-murray** [comment]
I am converting this issue to a discussion in case any further discussion is required

**Comment by sahidkhangithub** [comment]
@eric-murray , thanks for the reply.

As a service provider if we provide DeviceStatus service with input as just 1 field (msisdn) and output as fields mentioned in CAMARA swagger, then do it will be CAMARA approved api.

**Comment by eric-murray** [comment]
@sahidkhangithub 
Yes, that would compliant with the current API specification.

**Comment by Carmezim** [comment]
Are there plans to adopt OpenAPI v3.1 (added support for `oneOf` in parameters)? 
Could be a clearer option to document the endpoint parameters requirement on such use case.

**Comment by eric-murray** [comment]
@Carmezim 
Use of discriminators is allowed and is discussed [here](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md#1151-usage-of-discriminator) in the CAMARA API design guidelines. I don't know what additional benefit OAS v3.1 brings over v3.0.3 in this case - maybe you could elaborate.

But for the particular issue of identifying the device, the current specification states that "one or more" identifiers must be provided (`minProperties: 1`). Using a `oneOf` would require that exactly one device identifier must be provided. What is the benefit of that in this case? If the API consumer knows multiple identifiers for a given device, then allowing them to provide all of these identifiers increases the probability that the request can be fulfilled.
